syncer/syncer.go

```diff
@@ -1404,16 +1436,36 @@ func (s *Syncer) syncDML(
 
 // Run starts running for sync, we should guarantee it can rerun when paused.
 func (s *Syncer) Run(ctx context.Context) (err error) {
-	tctx := s.tctx.WithContext(ctx)
+	ctx2, cancel2 := context.WithCancel(context.Background())
+	tctx := s.tctx.WithContext(ctx2)
 
 	defer func() {
 		if s.done != nil {
 			close(s.done)
 		}
 	}()
 
+	go func() {
+		<-ctx.Done()
+		tctx.L().Info("received subtask's done")
+		if s.isTransactionEnd {
```

Comment:
About the assignment of `isTransactionEnd`, need to lock? @glorv @lance6716

Reply:
if there's concurrent accessing, we should use `atomic.Bool` at lease

Reply:
`atomic.Bool` is done